src/jit/hwintrinsicArm64.cpp

```diff
@@ -133,6 +133,60 @@ bool Compiler::impCheckImmediate(GenTree* immediateOp, unsigned int max)
     return immediateOp->IsCnsIntOrI() && (immediateOp->AsIntConCommon()->IconValue() < max);
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: gets the number of arguments for the hardware intrinsic.
+// This attempts to do a table based lookup but will fallback to the number
+// of operands in 'node' if the table entry is -1.
+//
+// Arguments:
+//    node      -- GenTreeHWIntrinsic* node with nullptr default value
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(GenTreeHWIntrinsic* node)
+{
+    NamedIntrinsic intrinsic = node->gtHWIntrinsicId;
+
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+
+    GenTree* op1     = node->gtGetOp1();
+    GenTree* op2     = node->gtGetOp2();
+    int      numArgs = 0;
+
+    if (op2 != nullptr)
+    {
+        return 2;
```

Comment:
You may want to add:

`assert(!op1->OperIsList());`

Reply:
Are the scenarios where `op1->OperIsList()` and `op2 != nullptr`?

Reply:
No, if `op1` is a list, `op2` must be null.